composeApp/src/androidMain/kotlin/io/github/couchtracker/ui/screens/main/MainScreen.kt

```diff
@@ -36,9 +36,12 @@ val MOVIE_COLOR_SCHEME = Color.hsv(180f, 1f, 0.5f).generateColorScheme()
 
 @Composable
 fun MainScreen() {
-    var currentSection by rememberSaveable { mutableStateOf(Section.HOME) }
     val insetTop = ScaffoldDefaults.contentWindowInsets.only(WindowInsetsSides.Top)
-    MaterialTheme(colorScheme = currentSection.mainColor) {
+    val navController = rememberNavController()
+    val currentSection = navController.currentBackStackEntryAsState().value?.destination?.route?.let { Section.fromId(it) } ?: Section.HOME
```

Comment:
This is a mouthful. Plus there a bunch null-safe calls as well. Can this be done in a cleaner way?